ext/openssl/ossl_pkey.c

```diff
@@ -628,6 +628,75 @@ ossl_pkey_initialize_copy(VALUE self, VALUE other)
 }
 #endif
 
+/*
+ *  call-seq:
+ *      OpenSSL::PKey.private_new(algo, string) -> PKey
+ *
+ * See the OpenSSL documentation for EVP_PKEY_new_raw_private_key()
+ */
+
+static VALUE
+ossl_pkey_initialize_private(VALUE self, VALUE type, VALUE key)
+{
+
+    EVP_PKEY *pkey;
+    const EVP_PKEY_ASN1_METHOD *ameth;
+    int pkey_id;
+    size_t keylen;
+
+#ifndef HAVE_EVP_PKEY_NEW_RAW_PRIVATE_KEY
```

Comment:
1. This doesn't cover the call of EVP_PKEY_new_raw_private_key below so this doesn't prevent a compile error when !HAVE_EVP_PKEY_NEW_RAW_PRIVATE_KEY.
2. [I believe] This kind of error should result into NoMethodError by excluding whole method implementation and declaration.

Reply:
Thanks for the comment!
The approach in 2. looks good so I went with that in [80e9ef1](https://github.com/ruby/openssl/pull/646/commits/80e9ef1a188bf8d712c976e622b165d330998b5c).